### PR TITLE
move preprocess callbacks inside model.preprocess (#2772)

### PR DESCRIPTION
**Pull request by jeffreyftang**
Cherry-pick #2772 into release branch.

### CODE REVIEW

**Comment by arnavgarg1** [comment]
@jeffreyftang Was this a cherry-pick after creating a branch from `release-0.6`? I'm seeing most of the tests fail because of the config_dict object, which I _believe_ didn't make it to `release-0.6` yet. cc: @connor-mccorm

**Comment by connor-mccorm** [comment]
Correct, we don't have the config object code in the 0.6 release yet. I believe we were planning on moving it into a 0.7 release so may need to wait for that unless we want to pull it into a minor release to unblock this.

**Review by tgaddair — APPROVED** [review]

**Pull request merged**
